service/worker/archiver/client.go

```diff
@@ -148,12 +148,13 @@ func NewClient(
 
 // Archive starts an archival task
 func (c *client) Archive(ctx context.Context, request *ClientRequest) (*ClientResponse, error) {
+	scopeWithDomainTag := c.metricsClient.Scope(metrics.ArchiverClientScope, metrics.DomainTag(request.ArchiveRequest.DomainName))
```

Comment:
There is a better approach. Check this method: https://sourcegraph.uberinternal.com/github.com/uber/cadence/-/blob/common/metrics/scope.go#L122

Reply:
You mean like using Tagged function, right? For example- 
c.metricsScope.Tagged(metrics.DomainTag(request.ArchiveRequest.DomainName))
instead of all other changes.

Reply:
yes